contracts/Cashier.sol

```diff
@@ -826,28 +841,158 @@ contract Cashier is ICashier, UsingHelpers, ReentrancyGuard, Ownable, Pausable {
      * @param _account  The address of an account to query
      * @param _newAmount  New amount to be set
      */
-    function updateEscrowTokensAmount(
+    function addEscrowTokensAmount(
         address _token,
         address _account,
         uint256 _newAmount
     ) external override onlyFromRouter {
-        escrowTokens[_token][_account] = _newAmount;
+        escrowTokens[_token][_account] =  escrowTokens[_token][_account].add(_newAmount);
     }
 
     /**
-     * @notice Update the amount in escrowTokens of an address with the new value, based on VoucherSet/Voucher interaction
-     * @param _token  The address of a token to query
-     * @param _account  The address of an account to query
-     * @param _newAmount  New amount to be set
+     * @notice Hook which will be triggered when a _tokenIdVoucher will be transferred. Escrow funds should be allocated to the new owner.
+     * @param _from prev owner of the _tokenIdVoucher
+     * @param _to next owner of the _tokenIdVoucher
+     * @param _tokenIdVoucher _tokenIdVoucher that has been transferred
      */
-    function addEscrowTokensAmount(
-        address _token,
-        address _account,
-        uint256 _newAmount
-    ) external override onlyFromRouter {
-        escrowTokens[_token][_account] =  escrowTokens[_token][_account].add(_newAmount);
+    function onERC721Transfer(
```

Comment:
My understanding was that we move the hooks in a separate contract, responsible only for those validations? Why we need to extend the Cashier?

Reply:
The hooks were making a lot of calls to the Cashier. Some calls are being made to VouherKernel as well, but most of the external calls were made to Cashier. By moving the hooks to Cashier, the functions can directly acces `escrowTokens` and `escrow` mappings. It removes several external calls per hook function, which makes the hook functions more efficient

Reply:
Somewhat agree, but don't feel this is the correct place. You are correct we are making calls to Cashier and VK, but for me we should have dedicated contract responsible for the hooks. @ali2251 Do you have an opinion here?

Reply:
I think if we reach a point where we are out of space in Cashier, we should create a new contract and move these there but for now it seems the easiest to do it this way